docs/core/magnetic_field.md

```diff
@@ -0,0 +1,228 @@
+arbitrary magnetic field configurations in an experiment. It is implemented in
+a generic way and can be extended to connect to an experiment specific upstream
+source of field data.
+
+Algorithms which need magnetic field information (e.g.
+{class}`Acts::AtlasStepper`, {class}`Acts::EigenStepper`) accept the magnetic
+field as an explicit argument. 
+
+## Provider interface
+
+All magnetic field implementations inherit and implement the interface {class}`Acts::MagneticFieldProvider`:
+
+
+It provides a generic interface over different implementations. To speed up
+magnetic field lookup, each implementation can have a *cache* object, which the
+client is expected to pass into lookup calls:
+
+:::{doxygentypedef} Acts::MagneticFieldProvider::Cache
+:::
+
+The implementation is then free to use and update this cache instance as
+needed. Before a client can issue field lookup calls, it needs to obtain an
+initialized instance of this cache object. This can be achieved generically for
+all implementations by using {func}`Acts::MagneticFieldProvider::makeCache`.
+This function accepts an instance of {type}`Acts::MagneticFieldContext`, see
+[](#magnetic-field-context) for details.
+
+The main lookup method of {class}`Acts::MagneticFieldProvider` is
+
+:::{doxygenfunction} Acts::MagneticFieldProvider::getField
+:outline:
+:::
+
+Aside from the lookup position
+as a global vector, it accepts an instance of the opaque cache object mentioned
+before. The return value is a {class}`Acts::Result` object. It either contains
+the field value at the requested location, or an
+{enum}`Acts::MagneticFieldError` in case of a lookup failure, like an
+out-of-bounds lookup position.
+
+Below is an example of how a client can interact with an instance of {class}`Acts::MagneticFieldProvider`.
+
+```cpp
+// in event context
+auto fieldContext = getExperimentFieldContext();
+const Acts::MagneticFieldProvider& fieldProvider = getFieldProvider();
+auto cache = fieldProvider.makeCache(fieldContext);
+
+auto lookupResult = fieldProvider.getField(Acts::Vector3{10, 10, 10}, cache);
+if(!lookupResult.ok()) {
+   throw std::runtime_error{"Field lookup failure"};
+}
+
+Acts::Vector3 fieldValue = *lookupResult;
+```
+
+## Magnetic field context
+
+:::{doxygentypedef} Acts::MagneticFieldContext
+:outline:
+:::
+
+The magnetic field context is an opaque type which contains experiment specific
+event context information. This can be used to supply event dependent data to
+the magnetic field instance, in case it is needed to provide correct field
+values. The library itself does not make any assumptions on the content of this
+context type (it is implemented using `std::any`), but passes a reference
+through the call-chain to the field implementation. An experiment specific
+field implementation is then expected to performa cast to the concrete type,
+and use the contents. 
+
+An example of this is a conditions database handle that is derived from the
```

Comment:
This sounds very Atlas. Maybe?:
An example of use of context could be to look up conditions database/records for the value of magnetic field in the time wen event was taken.